Assets/Scripts/Game/StartUp.cs

```diff
@@ -54,26 +56,33 @@ private void Start()
                 var playerData1 = playerFactory.CreatePlayerData(player1);
                 var playerData2 = playerFactory.CreatePlayerData(player2);
 
+                playerData1.Health = playerData2.Health = _maxHealth;
```

Comment:
<img width="382" alt="image" src="https://github.com/DariaDostoevskaia/Space/assets/44869070/796ad18a-be70-4251-8036-95389dbddc7e">
Try to convert this code through an array.
Those. initially create a new array of players based on how many GameContext.CurrentGameData.PlayersData.Count you have
If GameContext.CurrentGameData.PlayersData.Count == 0, then we consider this as the start of a new game